lib/internal/test_runner/test.js

```diff
@@ -552,23 +552,20 @@ class Suite extends Test {
     try {
       const context = { signal: this.signal };
       this.buildSuite = this.runInAsyncScope(this.fn, context, [context]);
+      if (isPromise(this.buildSuite)) {
+        this.buildSuite = this.buildSuite.catch((err) => {
+          this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
+        });
+      }
```

Comment:
Can we add a test with a non-Promise thenable? I think the following would work as expected on `main` but fail with this PR.
```js
test('successful thenable', () => {
  let thenCalled = false;
  return {
    get then() {
      if (thenCalled) throw new Error
      thenCalled = true;
      return (thenHandler) => successHandler();
    },
  };
);

test('rejected thenable', () => {
  let thenCalled = false;
  return {
    get then() {
      if (thenCalled) throw new Error
      thenCalled = true;
      return (_, errorHandler) => errorHandler(new Error);
    },
  };
);
```

Instead it should probably look like this:
```suggestion
      const then  = this.buildSuite?.then;
      if (typeof then === 'function') {
        this.buildSuite = ReflectApply(then, this.buildSuite, [undefined, (err) => {
          this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
        }]);
      }
```

Reply:
So should we add those tests? Otherwise it's likely to be missed in the future.

Reply:
done